txp/Pos/Txp/Toil/Utxo/Functions.hs

```diff
@@ -203,6 +204,8 @@ verifyKnownInputs VTxContext {..} resolvedInputs TxAux {..} = do
         ScriptWitness{..} -> do
             let valVer = scrVersion twValidator
                 redVer = scrVersion twRedeemer
+            when (valVer > vtcAdoptedScriptVersion) $
+                throwError $ WitnessAdoptedScriptVerMismatch valVer vtcAdoptedScriptVersion
             when (valVer /= redVer) $
                 throwError $ WitnessScriptVerMismatch valVer redVer
             when (vtcVerifyAllIsKnown && not (isKnownScriptVersion valVer)) $
```

Comment:
Can you clarify? Ideally this order shouldn't matter for your test. Your test should be done in such way that all conditions are satisfied except the one about adopted script version.
AFAIU, currently your test passes (because CI passes). Which permutations of these lines lead to failing tests?

Reply:
> Ideally this order shouldn't matter for your test
That is why I say it sucks. If you put `when (valVer /= redVer) $` line before `when (valVer > vtcAdoptedScriptVersion) $` then tests will fail.

Reply:
> If you put when (valVer /= redVer) $ line before when (valVer > vtcAdoptedScriptVersion) $ then tests will fail.

That's exactly what I wanted to know. It should be investigated, I'll take a look.

Reply:
Oh, well, that's because you use `goodIntRedeemer` as redeemer and it has version 0. Redeemer and validator should have same versions.

Reply:
Ok I will take a look and fix later today, thanks!